File: ourppc/SmartSearch/Form1.vb
Imports System.Threading

'Imports mshtml
Public Class Form1

    Public GoogleStep As Integer
    Public MaxIPercent As Double
    Public MaxKeyword As String
    Public BP, PB As Integer
    Public LastDoTime As DateTime
    Public ExeCount As Integer

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        LastDoTime = Now
        Me.WebBrowser1.Navigate("http://www.ourppc.com/tw/account-login")
        BP = 0
        PB = 0
        MaxKeyword = ""
        MaxIPercent = 0
        'GoogleStep = 0
        GoogleStep = 1
        'MsgBox(Me.WebBrowser1.Version.ToString)
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        'WebBrowser1.Document.GetElementById("yschsp").SetAttribute("value", Me.TextBox2.Text)
        'WebBrowser1.Document.GetElementById("sf").InvokeMember("submit")
        If ExeCount >= 3 Then
            Exit Sub
        End If
        Me.WebBrowser1.Navigate("http://www.ourppc.com/tw/p/ads")
        GoogleStep = 2
        
    End Sub

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        GoogleStep = 0
        Me.ComboBox1.SelectedIndex = 1
        LastDoTime = Now
    End Sub

    Private Sub Button3_Click(sender As Object, e As EventArgs) Handles Button3.Click
        Dim i As Integer
        Dim GetData As Boolean
        GetData = False
        Dim element As HtmlElement
        Dim elementList As HtmlElementCollection

        If ExeCount >= 3 Then
            Exit Sub
        End If

        element = Me.WebBrowser1.Document.GetElementById("sb-site").GetElementsByTagName("header").Item(0).GetElementsByTagName("section").Item(1).GetElementsByTagName("div").Item(0).GetElementsByTagName("div").Item(0).GetElementsByTagName("div").Item(1).GetElementsByTagName("li").Item(1).GetElementsByTagName("span").Item(1)
        Dim strBP = element.InnerHtml.Replace("$", "").Replace(",", "").Replace(".00", "")
        BP = CInt(strBP)

        element = Me.WebBrowser1.Document.GetElementById("sb-site").GetElementsByTagName("header").Item(0).GetElementsByTagName("section").Item(1).GetElementsByTagName("div").Item(0).GetElementsByTagName("div").Item(0).GetElementsByTagName("div").Item(1).GetElementsByTagName("li").Item(5).GetElementsByTagName("span").Item(1)
        Dim strPB = element.InnerHtml.Replace("$", "").Replace(",", "").Replace(".00", "")
        PB = CInt(strPB)
        If PB > 0 Then
            BP = PB
        End If
        Me.TextBox9.Text = BP.ToString
        
        If Me.CheckBox1.Checked = True Then
            Me.TextBox8.Text = BP.ToString
        End If
        Me.TextBox6.Text = BP.ToString + vbCrLf + Me.TextBox6.Text
        
        'WebBrowser1.Document.GetElementById("keyword").SetAttribute("Value", "e")
        WebBrowser1.Document.GetElementById("keyword").GetElementsByTagName("option").Item(Me.ComboBox1.SelectedIndex).SetAttribute("selected", "selected")
        'WebBrowser1.Document.GetElementById("keyword").InvokeMember("onchange")
        Dim seltype As String
        Select Case Me.ComboBox1.Text
            Case "標準"
                seltype = "type_s"
            Case "高級"
                seltype = "type_p"
            Case "專屬"
                seltype = "type_e"
            Case "專屬共享"
                seltype = "type_l"
        End Select
       
        Me.WebBrowser1.Document.GetElementById("type_s").Style = "display: none;"
        Me.WebBrowser1.Document.GetElementById("type_p").Style = "display: none;"
        Me.WebBrowser1.Document.GetElementById("type_e").Style = "display: none;"
        Me.WebBrowser1.Document.GetElementById("type_l").Style = "display: none;"
        Me.WebBrowser1.Document.GetElementById("" + seltype.ToString).Style = "display: ;"

        elementList = Me.WebBrowser1.Document.GetElementById(seltype).GetElementsByTagName("div")
        For Each element In elementList
            i = i + 1
            Debug.Print(i.ToString + ":::" + element.GetAttribute("classname").ToString)
            'Debug.Print(element.InnerHtml)
            If element.TagName() = "DIV" And element.GetAttribute("classname").ToString = "data-container" Then

                If InStr(element.GetElementsByTagName("header").Item(0).InnerHtml, Me.ComboBox1.Text) > 0 Then
                    For j = 1 To element.GetElementsByTagName("table").Item(0).GetElementsByTagName("tbody").Item(0).GetElementsByTagName("tr").Count
                        With element.GetElementsByTagName("table").Item(0).GetElementsByTagName("tbody").Item(0).GetElementsByTagName("tr").Item(j - 1).GetElementsByTagName("td")
                            If .Item(0).GetElementsByTagName("a").Item(0).GetAttribute("classname").ToString = "btn btn-sm btn-success make_invest" Then
                                ' Me.TextBox6.Text = Me.TextBox6.Text + .Item(0).InnerText + "   ,  "
                                'Me.TextBox6.Text = Me.TextBox6.Text + .Item(1).InnerText + "   ,  "
                                'Me.TextBox6.Text = Me.TextBox6.Text + .Item(2).InnerText + "   ,  "
                                'Me.TextBox6.Text = Me.TextBox6.Text + .Item(3).InnerText + "   ,  "


                                Dim strseday = .Item(4).GetElementsByTagName("span").Item(0).InnerText.Replace(" ", "")
                                strseday = strseday.Replace("年", "/")
                                strseday = strseday.Replace("月", "/")
                                strseday = strseday.Replace("日", "")
                                Dim arrayseday = strseday.Split("-")
                                Dim seday = DateTime.Parse(arrayseday(1)) - DateTime.Parse(arrayseday(0))
                                Dim iday = .Item(5).InnerText.Replace("天", "")
                                iday = iday.Replace(" ", "")
                                Dim stripercent = .Item(6).InnerText.Replace("%", "")
                                stripercent = stripercent.Replace(" ", "")
                                Dim arrayipercent = stripercent.Split("-")
                                Dim ipercent = ((arrayipercent(1) + 0) + (arrayipercent(0) + 0)) / 2
                                If (ipercent / (seday.Days + iday)) > MaxIPercent Then
                                    MaxIPercent = (ipercent / (seday.Days + iday))
                                    MaxKeyword = .Item(1).InnerText
                                End If
                                ' Me.TextBox6.Text = (ipercent / (seday.Days + iday)).ToString + "   ,  " + Me.TextBox6.Text

                                'Me.TextBox6.Text = Me.TextBox6.Text + vbCrLf
                            End If
                        End With
                    Next
                    GoogleStep = 3
                    Button4_Click(Me, e)
                    Me.TextBox1.Text = "Button4_Click(Me, e)" + Me.TextBox1.Text + vbCrLf
                End If
                Me.TextBox6.Text = MaxKeyword + "::::" + MaxIPercent.ToString + vbCrLf + Me.TextBox6.Text
                'Me.TextBox6.Text = Me.TextBox6.Text + ":::::" + elementList(i - 1).InnerHtml
                'For Each element2 As HtmlElement In elementList
                'Debug.Print(element2.GetAttribute("classname").ToString)
                'Next
            End If
        Next
        ''If GetData = False Then
        ''    'Call Button6_Click(Me, e)
        ''    GoogleStep = 1
        ''End If
    End Sub

    Private Sub Button4_Click(sender As Object, e As EventArgs) Handles Button4.Click
        Dim element, element2 As HtmlElement
        Dim elementList, elementList2 As HtmlElementCollection

        If ExeCount >= 3 Then
            Exit Sub
        End If
        ExeCount = ExeCount + 1
        If MaxIPercent < CDbl(Me.TextBox7.Text) Then
            GoogleStep = 1
            Me.TextBox1.Text = "MaxIPercent < Me.TextBox7.Text" + ";" + Me.TextBox1.Text
            Button2_Click(Me, e)
            Exit Sub
        End If
        If BP <= 0 Then
            GoogleStep = 1
            Me.TextBox1.Text = GoogleStep.ToString + ";" + Me.TextBox1.Text
            Exit Sub
        End If
        Dim seltype
        Select Case Me.ComboBox1.Text
            Case "標準"
                seltype = "type_s"
            Case "高級"
                seltype = "type_p"
            Case "專屬"
                seltype = "type_e"
            Case "專屬共享"
                seltype = "type_l"
        End Select
        elementList = Me.WebBrowser1.Document.GetElementById(seltype.ToString).GetElementsByTagName("div")
        For Each element In elementList
            'Debug.Print(element.InnerHtml)
            If element.TagName() = "DIV" And element.GetAttribute("classname").ToString = "data-container" Then
                If InStr(element.GetElementsByTagName("header").Item(0).InnerHtml, Me.ComboBox1.Text) > 0 Then
                    For j = 1 To element.GetElementsByTagName("table").Item(0).GetElementsByTagName("tbody").Item(0).GetElementsByTagName("tr").Count
                        With element.GetElementsByTagName("table").Item(0).GetElementsByTagName("tbody").Item(0).GetElementsByTagName("tr").Item(j - 1).GetElementsByTagName("td")
                            If .Item(0).GetElementsByTagName("a").Item(0).GetAttribute("classname").ToString = "btn btn-sm btn-success make_invest" Then
                                If MaxKeyword = .Item(1).InnerText Then
                                    .Item(0).GetElementsByTagName("a").Item(0).Id = "gggggggg"
                                    WebBrowser1.Document.GetElementById("gggggggg").InvokeMember("click")
                                    System.Threading.Thread.Sleep(0.3)
                                    WebBrowser1.Document.GetElementById("amount").SetAttribute("value", Me.TextBox8.Text)
                                    elementList2 = Me.WebBrowser1.Document.GetElementsByTagName("form")
                                    For Each element2 In elementList2
                                        If element2.GetAttribute("name").ToString = "invest_ads_form" Then
                                            element2.InvokeMember("submit")
                                            MaxIPercent = 0
                                            MaxKeyword = ""
                                            BP = 0
                                            GoogleStep = 1
                                        End If
                                    Next
                                    '.Item(0).GetElementsByTagName("a").Item(0).Focus()
                                    'SendKeys.Send("{ENTER}")
                                End If
                            End If
                        End With
                    Next
                End If
            End If
        Next
        'WebBrowser1.Document.GetElementById("pg-next").Focus()
        'WebBrowser1.Document.GetElementById("pg-next").InvokeMember("click")
        'Me.TextBox4.Text = Me.TextBox4.Text + 1
    End Sub

    Private Sub Button5_Click(sender As Object, e As EventArgs)
        Me.WebBrowser1.Navigate("https://www.google.com.tw/")
    End Sub

    Private Sub Button8_Click(sender As Object, e As EventArgs)
        WebBrowser1.Document.GetElementById("lst-ib").SetAttribute("value", Me.TextBox2.Text)
        WebBrowser1.Document.GetElementById("tsf").InvokeMember("submit")

        Me.Timer1.Enabled = True
    End Sub

    Private Sub Button6_Click(sender As Object, e As EventArgs)

        WebBrowser1.Document.GetElementById("pnnext").Focus()
        WebBrowser1.Document.GetElementById("pnnext").InvokeMember("click")
        Me.TextBox4.Text = Me.TextBox4.Text + 1
        GoogleStep = 2
        ' Dim rndNum As New Random()
        ' Dim ss = rndNum.Next(1, 5)
        'Thread.Sleep(ss * 1000)
        'Me.TextBox1.Text = Me.TextBox1.Text + vbCrLf + ss.ToString
        'Call Button7_Click(Me, e)
    End Sub

    Private Sub Button7_Click(sender As Object, e As EventArgs)

        Dim i As Integer
        Dim GetData As Boolean
        GetData = False
        For Each element As HtmlElement In Me.WebBrowser1.Document.All
            i = i + 1
            If element.TagName() = "A" And element.GetAttribute("onmousedown").ToString <> "" Then
                If element.GetAttribute("href").ToString = Me.TextBox3.Text Then
                    Dim startpos = InStr(1, element.OuterHtml.ToString(), "return rwt(this,'','','','")
                    If startpos > 0 Then
                        startpos = startpos + 26
                        Dim endpos = InStr(startpos, element.OuterHtml.ToString(), "'")
                        Me.TextBox5.Text = element.OuterHtml.Substring(startpos - 1, endpos - startpos)
                        'Me.TextBox1.Text = Me.TextBox1.Text + element.OuterHtml.ToString()
                        'onmousedown="return rwt(this,'','','','
                        Me.TextBox1.Text = Me.TextBox1.Text + "第" + Me.TextBox4.Text + "頁,第" + Me.TextBox5.Text + "筆" + ":::" + element.GetAttribute("href").ToString + vbCrLf
                        GetData = True
                        GoogleStep = 99
                    End If
                End If
            End If
        Next
        If GetData = False Then
            'Call Button6_Click(Me, e)
            GoogleStep = 1
        End If
    End Sub

    Private Sub Timer1_Tick(sender As Object, e As EventArgs) Handles Timer1.Tick
        If DateDiff("s", LastDoTime, Now) > 180 Then
            Button1_Click(Me, e)
        End If
        ''Select Case GoogleStep
        ''    Case 0
        ''        Call Button7_Click(Me, e)
        ''    Case 1
        ''        Call Button6_Click(Me, e)
        ''    Case 2
        ''        Call Button7_Click(Me, e)
        ''End Select
        ''If Me.TextBox4.Text > 10 Then
        ''    Me.Timer1.Enabled = False
        ''End If
    End Sub

    Private Sub WebBrowser1_DocumentCompleted(sender As Object, e As WebBrowserDocumentCompletedEventArgs) Handles WebBrowser1.DocumentCompleted       
        LastDoTime = Now
        Select Case GoogleStep
            Case 0
                ' ''If e.Url.ToString = "http://www.ourppc.com/tw/p/" Then
                ' ''    GoogleStep = 1
                ' ''    Button2_Click(Me, e)
                ' ''    Exit Sub
                ' ''End If
                ' ''If InStr(e.Url.ToString, "http://www.ourppc.com/tw/account-login") > 0 Then
                ' ''    Me.TextBox1.Text = e.Url.ToString + vbCrLf + Me.TextBox1.Text
                ' ''    Dim captcha As HtmlElement = WebBrowser1.Document.GetElementById("captcha")
                ' ''    '判斷captcha是否存在
                ' ''    If captcha IsNot Nothing Then
                ' ''        '調整顯示的位置
                ' ''        captcha.ScrollIntoView(False)
                ' ''        Try
                ' ''            'WebBrowser1.Document.DomDocument.GetType().GetProperty("designMode").SetValue(WebBrowser1.Document.DomDocument, "On", Nothing)
                ' ''            'Me.WebBrowser1.Document.ExecCommand.ExecCommand("EditMode", True, Nothing)
                ' ''            Dim doc As IHTMLDocument2 = DirectCast(WebBrowser1.Document.DomDocument, IHTMLDocument2)
                ' ''            Dim body As IHTMLElement2 = DirectCast(doc.body, IHTMLElement2)
                ' ''            Dim rang As IHTMLControlRange = DirectCast(body.createControlRange(), IHTMLControlRange)
                ' ''            Dim Img As IHTMLControlElement = DirectCast(DirectCast(doc, IHTMLDocument3).getElementById("captcha"), IHTMLControlElement)
                ' ''            ''imgRange.add(image)
                ' ''            ''imgRange.execCommand("Copy", False, Nothing)


                ' ''            'Dim rang = WebBrowser1.Document.DomDocument.Body.createControlRange()
                ' ''            Me.TextBox1.Text = rang.ToString + vbCrLf + Me.TextBox1.Text
                ' ''            ''Dim Img = captcha.DomElement
                ' ''            Me.TextBox1.Text = "000000" + vbCrLf + Me.TextBox1.Text
                ' ''            rang.add(Img)
                ' ''            Me.TextBox1.Text = "11111111" + vbCrLf + Me.TextBox1.Text
                ' ''            rang.execCommand("Copy", False, Nothing)
                ' ''            Dim RegImg As Image = Clipboard.GetImage()
                ' ''            Clipboard.Clear()
                ' ''            Me.BackgroundImageLayout = ImageLayout.None
                ' ''            Me.BackgroundImage = RegImg

                ' ''            Me.BackgroundImage.Save(Application.StartupPath.ToString + "\test_save.bmp")

                ' ''            '取得captcha圖片的src
                ' ''            Dim src As String = Application.StartupPath.ToString + "\test_save.bmp"
                ' ''            '從src取得圖片
                ' ''            Dim img2 As New Bitmap(GetImageFromURL(src))
                ' ''            '進行OCR並將結果填到網頁中
                ' ''            Dim element As IHTMLElement
                ' ''            Dim elementList As IHTMLElementCollection

                ' ''            elementList = DirectCast(doc, IHTMLDocument3).getElementsByTagName("input")
                ' ''            For Each element In elementList
                ' ''                'Me.TextBox1.Text = "eee:" + element.getAttribute("name").ToString + vbCrLf + Me.TextBox1.Text
                ' ''                If element.getAttribute("name").ToString = "captcha" Then
                ' ''                    element.setAttribute("value", TesseractOCR(img2))
                ' ''                End If
                ' ''            Next
                ' ''            DirectCast(doc, IHTMLDocument3).getElementById("username").setAttribute("value", Me.TextBox2.Text)
                ' ''            DirectCast(doc, IHTMLDocument3).getElementById("password").setAttribute("value", Me.TextBox3.Text)
                ' ''            DirectCast(doc, IHTMLDocument3).getElementById("login_btn").click()
                ' ''            'WebBrowser1.Document.GetElementById("password").SetAttribute("value", Me.TextBox3.Text)
                ' ''            'WebBrowser1.Document.GetElementById("login_btn").InvokeMember("click")
                ' ''        Catch ex As Exception
                ' ''            MessageBox.Show(ex.Message)
                ' ''        End Try
                ' ''    End If
                ' ''End If
            Case 1
                Button2_Click(Me, e)
                Me.TextBox1.Text = "Button2_Click(Me, e)" + vbCrLf + Me.TextBox1.Text
            Case 2
                Button3_Click(Me, e)
                Me.TextBox1.Text = "Button3_Click(Me, e)" + vbCrLf + Me.TextBox1.Text
            Case 3
                GoogleStep = 1
                Button2_Click(Me, e)
                Me.TextBox1.Text = "Button2_Click(Me, e)" + vbCrLf + Me.TextBox1.Text
        End Select
        
    End Sub

    Private Sub WebBrowser1_StatusTextChanged(sender As Object, e As EventArgs) Handles WebBrowser1.StatusTextChanged
        ''On Error Resume Next
        ''Dim head As HtmlElement = WebBrowser1.Document.GetElementsByTagName("head")(0)
        ''Dim script As HtmlElement
        ''script = WebBrowser1.Document.CreateElement("script")
        ''script.SetAttribute("type", "text/javascript")
        ''Dim alertBlocker As String = "window.alert = function () { }; window.confirm=function () { }; "
        ''script.SetAttribute("text", alertBlocker)
        ''head.AppendChild(script)
    End Sub

    Private Sub Button5_Click_1(sender As Object, e As EventArgs) Handles Button5.Click
        Me.WebBrowser1.Navigate("http://my-user-agent.com/")

    End Sub

    Private Sub Button6_Click_1(sender As Object, e As EventArgs) Handles Button6.Click
        Me.TextBox1.Text = GoogleStep.ToString + ":" + BP.ToString + ":" + PB.ToString + ";" + Me.TextBox1.Text + vbCrLf

    End Sub

    Private Sub Button7_Click_1(sender As Object, e As EventArgs) Handles Button7.Click
        GoogleStep = -1
    End Sub

    Function TesseractOCR(img As Bitmap) As String
        Try
            Dim ocr As New tessnet2.Tesseract
            ocr.SetVariable("tessedit_char_whitelist", "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
            ocr.Init(".\tessdata", "eng", False)
            Dim result As List(Of tessnet2.Word) = ocr.DoOCR(img, Rectangle.Empty)
            Dim str As String = ""
            For i As Integer = 0 To result.Count - 1
                str &= result(i).Text
            Next
            Return str
        Catch ex As Exception
            Throw ex
        End Try
        Return String.Empty
    End Function

    Function GetImageFromURL(url As String) As Image
        Try

            Dim fs As New System.IO.FileStream(url, IO.FileMode.Open)
            Dim img As Image = Image.FromStream(fs)
            fs.Close()
            'Dim request As Net.WebRequest = Net.WebRequest.Create(url)
            'Dim response As Net.WebResponse = request.GetResponse()
            'Dim stream As IO.Stream = response.GetResponseStream()
            'Dim img As New Bitmap(stream)
            Return img
        Catch ex As Exception
            Throw ex
        End Try
        Return Nothing
    End Function


    Private Sub ComboBox1_SelectedIndexChanged(sender As Object, e As EventArgs) Handles ComboBox1.SelectedIndexChanged
        MaxIPercent = 0
        MaxKeyword = ""
        BP = 0
        PB = 0
    End Sub

    Private Sub Timer2_Tick(sender As Object, e As EventArgs) Handles Timer2.Tick

        Dim dt As String

        Try
            If InStr(Me.WebBrowser1.Document.Body.InnerHtml, "無法顯示") > 0 Then
                Me.TextBox1.Text = "網頁無法連線重置" + vbCrLf + Me.TextBox1.Text
                Button1_Click(Me, e)
                Exit Sub
            End If
            dt = Me.WebBrowser1.Document.GetElementById("local_time_text").InnerText.Substring(6, 2)
            Me.TextBox2.Text = "wb:" + WebBrowser1.StatusText + ",GS:" + GoogleStep.ToString + ",dt:" + dt.ToString + ",EC:" + ExeCount.ToString
            If Me.WebBrowser1.Url.ToString = "http://www.ourppc.com/tw/p/ads" Then
                If dt = Me.TextBox5.Text Then
                    ExeCount = 1
                    Button2_Click(Me, e)
                End If
            End If
        Catch ex As Exception

        End Try
        '
    End Sub

    Private Sub Button8_Click_1(sender As Object, e As EventArgs) Handles Button8.Click
        Me.TextBox1.Text = ""
        Me.TextBox6.Text = ""
    End Sub
End Class

File: get_web_table/SearchWeb/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Data.SqlClient;
using System.Configuration;
using System.Threading;

namespace SearchWeb
{
    public partial class Form1 : Form
    {
        int type = 0;
        int webtype = 1; //1=google 2=yahoo
        int searchtype = 1;
        int stop = 0;
        int index = 1;  //is page
        string[] substr;
        string nowwwstr = "";
        int button_pn;

        int searchcount = 0;

        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            panel1.Height = this.Height;
            button3.Visible = false;
            button4.Visible = false;
            button5.Visible = false;
            button6.Visible = false;
            button7.Visible = false;
            button8.Visible = false;
            button9.Visible = false;
            button10.Visible = false;
            textBox1.Visible = false;
            //label4.Visible = false;
            label3.Visible = false;
        }

        private void button1_Click(object sender, EventArgs e) 
        {
            if (searchcount < 100)  //從資料庫撈取總數
            {
                if (int.Parse(textBox4.Text) * 1000 < 10000)
                {
                    MessageBox.Show("時間設定請勿小於10秒");
                }
                else
                {
                    timer2.Interval = int.Parse(textBox4.Text) * 1000;
                    substr = textBox3.Text.Split('.');
                    for (int i = 1; i < substr.Length; i++)
                    {
                        if (i == 1)
                        {
                            nowwwstr += substr[i];
                        }
                        else
                        {
                            nowwwstr += "." + substr[i];
                        }
                    }

                    if (textBox2.Text != "" && textBox3.Text != "" && webtype == 1)
                    {
                        index = 1;
                        type = 1;
                        searchtype = 1;
                        stop = 0;
                        webBrowser1.Navigate("https://www.google.com.tw/");
                    }
                    else if (textBox2.Text != "" && textBox3.Text != "" && webtype == 2)
                    {
                        webtype = 2;
                        index = 1;
                        type = 1;
                        searchtype = 1;
                        button_pn = 0;
                        stop = 0;
                        webBrowser1.Navigate("https://tw.yahoo.com/");
                    }
                    else
                    {
                        MessageBox.Show("搜尋欄位及條件不可為空");
                    }
                    searchcount++;
                }
            }
            else
            {
                timer2.Enabled = true;
            }
        }

        private void webBrowser1_DocumentCompleted(object sender, WebBrowserDocumentCompletedEventArgs e)
        {
            if (type == 1)
            {
                button6_Click(this, e);
            }
            if (button_pn == 1)
            {
                button_pn = 0;
                button4_Click(this, e);
            }
            
        }

        private void SaveData(string web_page,string web_item,string web_type,string web_search,string web_condition)
        {
            string strconn = ConfigurationManager.ConnectionStrings["SearchWeb.Properties.Settings.websearchConnectionString"].ToString();
            SqlCommand cmd;

            SqlConnection sqlconn = new SqlConnection(strconn);
            sqlconn.Open();

            string sql = "insert into [websearch].[dbo].[search] (web_page,web_item,web_type,web_search,web_condition) values (@web_page,@web_item,@web_type,@web_search,@web_condition)";
            cmd = new SqlCommand(sql, sqlconn);
            cmd.Parameters.Add("@web_page", SqlDbType.NChar).Value = web_page;
            cmd.Parameters.Add("@web_item", SqlDbType.NChar).Value = web_item;
            cmd.Parameters.Add("@web_type", SqlDbType.NChar).Value = web_type;
            cmd.Parameters.Add("@web_search", SqlDbType.NChar).Value = web_search;
            cmd.Parameters.Add("@web_condition", SqlDbType.NChar).Value = web_condition;
            cmd.ExecuteNonQuery();

            sqlconn.Close();
            sqlconn.Dispose();
            cmd.Dispose();
        }

        private void button3_Click(object sender, EventArgs e)
        {
            HtmlElement element;
            if (webtype == 1)
            {
                element = webBrowser1.Document.GetElementById("tsf");
                element.InvokeMember("submit");
                timer1.Interval = 2000;
            }
            else if (webtype == 2)
            {
                element = webBrowser1.Document.GetElementById("UHSearch");
                element.InvokeMember("submit");
            }
            timer1.Enabled = true;

        }

        private async void button4_Click(object sender, EventArgs e)
        {
            int item = 0;  //第一項
            HtmlElementCollection elementlist;
            if (type == 0 && webtype == 1 && stop != 1)
            {
                elementlist = webBrowser1.Document.GetElementsByTagName("div");
                foreach (HtmlElement curhtml in elementlist)
                {
                    await Task.Delay(100);
                    label4.Text = curhtml.GetAttribute("classname").ToString();
                    if (curhtml.GetAttribute("classname").ToString() == "f kv _SWb")
                    {
                        item++;
                        //richTextBox1.AppendText(item.ToString()+" "+ curhtml.GetAttribute("classname").ToString() +"\n");
                        //label4.Text = item.ToString();
                        HtmlElement citehtml = curhtml.GetElementsByTagName("cite")[0];
                        label3.Text = item.ToString();
                        if (citehtml.GetAttribute("classname").ToString() == "_Rm")
                        {
                            if ((citehtml.InnerHtml.IndexOf(nowwwstr) >= 0) || (citehtml.InnerHtml.IndexOf(textBox3.Text) >= 0))
                            {
                                stop = 1; 
                                textBox1.Text = citehtml.InnerHtml;
                                SaveData(index.ToString(), item.ToString(), "google", textBox2.Text, textBox3.Text);
                                nowwwstr = "";
                                break;
                            }
                        }
                    }
                }
            }
            else if (type == 0 && webtype == 2 && stop != 1)
            {
                elementlist = webBrowser1.Document.GetElementsByTagName("ol");
                foreach (HtmlElement curhtml in elementlist)
                {
                    await Task.Delay(100);
                    if (curhtml.GetAttribute("classname").ToString() == " reg searchCenterMiddle")
                    {
                        HtmlElementCollection lihtmllist = curhtml.GetElementsByTagName("li");
                        foreach (HtmlElement lihtml in lihtmllist)
                        {
                            await Task.Delay(100);
                            item++;
                            HtmlElementCollection spanlist = lihtml.GetElementsByTagName("span");
                            foreach (HtmlElement spanhtml in spanlist)
                            {
                                if (spanhtml.GetAttribute("classname").ToString() == " fz-ms fw-m fc-12th wr-bw")
                                {
                                    if (spanhtml.InnerHtml.IndexOf(textBox3.Text) >= 0 || spanhtml.InnerHtml.IndexOf(nowwwstr) >= 0)
                                    {
                                        stop = 1;
                                        SaveData(index.ToString(), item.ToString(), "yahoo", textBox2.Text, textBox3.Text);
                                        nowwwstr = "";
                                        break;
                                    }
                                }
                            }
                            if (stop == 1)
                            {
                                break;
                            }
                        }
                    }
                    if (stop == 1)
                    {
                        break;
                    }
                }
            }

            if (webtype == 1 && stop != 1 && (index < 10))
            {   
                index++;
                await Task.Delay(200);
                button5_Click(this, e);
            }
            else if (webtype == 1 && stop != 1 && index >= 10)
            {
                webtype = 2;
                nowwwstr = "";
                button1_Click(this, e);
            }
            else if (webtype == 2 && stop != 1 && index < 10)
            {
                index++;
                await Task.Delay(200);
                button5_Click(this, e);
            }
            else if (webtype == 2 && stop != 1 && index >= 10)
            {
                stop = 1;
                webtype = 1;
                timer2.Enabled = true;
            }
            if ((stop == 1) && (webtype==1))
            {
                webtype = 2;
                button1_Click(this, e);
            }else if ((stop == 1) && (webtype==2))
            {
                webtype = 1;
                nowwwstr = "";
                timer2.Enabled = true;
            }

        }

        private void button5_Click(object sender, EventArgs e)
        {
            HtmlElementCollection elementlist;
            button_pn = 1;
            if (webtype == 1)
            {
                elementlist = webBrowser1.Document.GetElementsByTagName("a");
                foreach (HtmlElement curhtml in elementlist)
                {
                    if (curhtml.GetAttribute("id").ToString() == "pnnext")
                    {
                        curhtml.InvokeMember("click");
                        break;
                    }
                }
            }
            else if (webtype == 2)
            {
                elementlist = webBrowser1.Document.GetElementsByTagName("a");
                foreach (HtmlElement curhtml in elementlist)
                {
                    if (curhtml.GetAttribute("classname").ToString() == "next")
                    {
                        curhtml.InvokeMember("click");
                        break;
                    }
                }
            }
        }

        private void button2_Click(object sender, EventArgs e)
        {
            webBrowser1.Navigate("http://my-user-agent.com/");
            type = 0;
        }

        private void label1_Click(object sender, EventArgs e)
        {

        }

        private void timer1_Tick(object sender, EventArgs e)
        {
            if (searchtype == 1)
            {
                type = 0;
                timer1.Enabled = false;
                searchtype = 2;
                button3_Click(this, e);
            }
            else if (searchtype == 2)
            {
                timer1.Enabled = false;
                button4_Click(this, e);
            }      
        }

        private void button6_Click(object sender, EventArgs e)
        {
            type = 0;
            if (webtype == 1)
            {
                HtmlElement element;
                element = webBrowser1.Document.GetElementById("lst-ib");
                element.InnerText = textBox2.Text;
                searchtype = 1;
                timer1.Interval = 1000;
            }
            else if (webtype == 2)
            {
                button8_Click(this, e);
                timer1.Interval = 10000;
            }
            timer1.Enabled = true;
        }

        private void button7_Click(object sender, EventArgs e)
        {
            webBrowser1.Navigate("https://tw.yahoo.com/");
        }

        private async void button8_Click(object sender, EventArgs e)
        {
            await Task.Delay(2000);
            HtmlElement element;
            element = webBrowser1.Document.GetElementById("UHSearchBox");
            element.InnerText = textBox2.Text;
        }

        private void button9_Click(object sender, EventArgs e)
        {
            HtmlElementCollection elementlist;
            elementlist = webBrowser1.Document.GetElementsByTagName("a");
            foreach (HtmlElement curhtml in elementlist)
            {
                if (curhtml.GetAttribute("classname").ToString() == "next")
                {
                    curhtml.InvokeMember("click");
                    break;
                }
            }
        }

        private void button10_Click(object sender, EventArgs e)
        {
            HtmlElement element;
            element = webBrowser1.Document.GetElementById("UHSearch");
            element.InvokeMember("submit");
        }

        private void timer2_Tick(object sender, EventArgs e)
        {
            searchcount = 0;
            button1_Click(sender, e);
            timer2.Enabled = false;
        }

        private void textBox1_TextChanged(object sender, EventArgs e)
        {

        }
    }
}
